src/odemis/driver/test/pwrcomedi_test.py

```diff
@@ -65,61 +65,55 @@ def tearDown(self):
 
         # turn on second source to 90%
-        self.dev.power.value = self.dev.power.range[1]
-        em = [0, 0.9]
-        self.dev.emissions.value = em
-        self.assertGreater(self.dev.emissions.value[1], 0)
-
-    def test_multi(self):
-        """
-        simultaneous source activation
-        """
-        self.dev.power.value = self.dev.power.range[1]
-        em = [1] * len(self.dev.emissions.value)
-        self.dev.emissions.value = em
-        # They should all be on
-        self.assertTrue(all(e > 0 for e in self.dev.emissions.value))
-
-        # Not all should be at the max, due to clamping
-        self.assertTrue(any(e < 1 for e in self.dev.emissions.value))
-
+        self.dev.power.value[1] = self.dev.power.range[1][1] * 0.9
+        self.assertGreater(self.dev.power.value[1], 0)
+
+    # def test_multi(self):
+    #     """
+    #     simultaneous source activation
+    #     """
+    #     self.dev.power.value = self.dev.power.range[1]
+    #     em = [1] * len(self.dev.emissions.value)
+    #     self.dev.emissions.value = em
+    #     # They should all be on
+    #     self.assertTrue(all(e > 0 for e in self.dev.emissions.value))
+    #
+    #     # Not all should be at the max, due to clamping
+    #     self.assertTrue(any(e < 1 for e in self.dev.emissions.value))
+    #
     def test_cycle(self):
         """
         Test each emission source for 2 seconds at maximum intensity and then 1s
         at 30%.
         """
-        em = [0] * len(self.dev.emissions.value)
         self.dev.power.value = self.dev.power.range[1]
 
         # can fully checked only by looking what the hardware is doing
         logging.info("Starting emission source cycle...")
-        for i in range(len(em)):
+        for i in range(len(self.dev.power.value)):
             logging.info("Turning on wavelength %g", self.dev.spectra.value[i][2])
-            em[i] = 1
-            self.dev.emissions.value = em
+            self.dev.power.value[i] *= 1
+            time.sleep(1)
+            self.assertGreater(self.dev.power.value[i], 0)  # Can't check for equality due to clamping
+
+            self.dev.power.value[i] *= 0.3
             time.sleep(1)
-            self.assertGreater(self.dev.emissions.value[i], 0)  # Can't check for equality due to clamping
+            self.assertGreater(self.dev.power.value[i], 0)
 
-            em[i] = 0.3
-            self.dev.emissions.value = em
+            # value so small that it's == 0 for the hardware
+            self.dev.power.value[i] *= 1e-8
             time.sleep(1)
-            self.assertEqual(self.dev.emissions.value, em)
-#             # value so small that it's == 0 for the hardware
-#             self.dev.emissions.value[i] = 1e-8
-#             em[i] = 0
-#             self.assertEqual(self.dev.emissions.value, em)
+            self.assertGreater(self.dev.power.value[i], 0)
+            # self.assertEquals(self.dev.power.value[i], 0)
```

Comment:
How can we check it's so small that it's 0 for the driver? Currently it will always be greater by fraction.

Reply:
Indeed, the driver would need to reverse back the actual data into a power value to realise it's actually 0. Not a big deal though.